lib/cartopy/img_transform.py

```diff
@@ -276,17 +279,28 @@ def regrid(array, source_x_coords, source_y_coords, source_cs, target_proj,
                                            target_y_points.flatten())
 
     if _is_pykdtree:
-        kdtree = pykdtree.kdtree.KDTree(xyz)
+        cache_name = 'pykdtree'
```

Comment:
I don't understand this cache key; how is it supposed to work with multiple images?